### PR TITLE
refactor: change `Header#dao` to Byte32

### PR DESCRIPTION
**Pull request by quake**
[DONE] one integration test is  broken, waiting for https://github.com/nervosnetwork/ckb-system-scripts/pull/40

### CODE REVIEW

File: util/dao/utils/src/lib.rs [hunk 1]
```diff
@@ -58,22 +58,19 @@ pub fn genesis_dao_data(genesis_cellbase_tx: &TransactionView) -> Result<Bytes,
     Ok(pack_dao_data(DEFAULT_ACCUMULATED_RATE, c, u))
 }
 
-pub fn extract_dao_data(data: &[u8]) -> Result<(u64, Capacity, Capacity), FailureError> {
-    if data.len() != DAO_SIZE || data[0] != DAO_VERSION {
-        return Err(Error::Format.into());
-    }
```
Comment (u2) [inline util/dao/utils/src/lib.rs]: Please keep `data[0] != DAO_VERSION`.

Comment (quake) [reply util/dao/utils/src/lib.rs]: Resolved, please review again, thanks.

**Comment by doitian** [comment]
@quake ckb-system-scripts updated, please use `= 0.3.1`

File: resource/Cargo.toml [hunk 2]
```diff
@@ -13,10 +13,10 @@ tempfile = "3.0"
 serde = "1.0"
 serde_derive = "1.0"
 ckb-types = { path = "../util/types" }
-ckb-system-scripts = { version = "= 0.3.1-alpha.3" }
+ckb-system-scripts = { version = "= 0.3.1-alpha.4" }
```
Comment (doitian) [inline resource/Cargo.toml]:
```suggestion
ckb-system-scripts = "= 0.3.1"
```

File: resource/Cargo.toml [hunk 3]
```diff
@@ -13,10 +13,10 @@ tempfile = "3.0"
...
 
 [build-dependencies]
 includedir_codegen = "0.5.0"
 walkdir = "2.1.4"
 ckb-types = { path = "../util/types" }
-ckb-system-scripts = { version = "= 0.3.1-alpha.3" }
+ckb-system-scripts = { version = "= 0.3.1-alpha.4" }
```
Comment (doitian) [inline resource/Cargo.toml]:
```suggestion
ckb-system-scripts = "= 0.3.1"
```

File: benches/Cargo.toml [hunk 4]
```diff
@@ -26,7 +26,7 @@ ckb-occupied-capacity = { path = "../util/occupied-capacity"}
 ckb-test-chain-utils = { path = "../util/test-chain-utils" }
 ckb-dao-utils = { path = "../util/dao/utils" }
 ckb-dao = { path = "../util/dao" }
-ckb-system-scripts = { version = "0.3.1-alpha.3" }
+ckb-system-scripts = { version = "0.3.1-alpha.4" }
```
Comment (doitian) [inline benches/Cargo.toml]:
```suggestion
ckb-system-scripts = "= 0.3.1"
```

**Comment by quake** [comment]
> @quake ckb-system-scripts updated, please use `= 0.3.1`

Updated, please review again, thanks

**Comment by u2** [comment]
> > @quake ckb-system-scripts updated, please use `= 0.3.1`
> 
> Updated, please review again, thanks

ckb-system-scripts = "= 0.3.1"

**Comment by u2** [comment]
bors r=doitian,u2

**Comment by quake** [comment]
bors r=doitian,u2

**Pull request merged**